Fixing "Add Tag" command for Windows and Updating Mitre Techniques

msoheem opened the pull request:
Made two changes:

1. The tag addition command was incorrectly placing tags on Windows systems due to differences in line endings. The underlying regular expression has been adjusted to ensure accurate tag placement on both Linux and Windows.
![image](https://github.com/user-attachments/assets/5a78f7aa-451c-4f38-937f-b50f48ead07f)


2. Updated MITRE Enterprise ATT&CK Techniques as i found that current database has deprecated IDs such as T1050. Source of file https://attack.mitre.org/docs/enterprise-attack-v16.1/enterprise-attack-v16.1-techniques.xlsx

msoheem committed:
Update commandProvider.ts

Updated RegExp for finding "tags:" in the active editor as it works well for Linux system however, it does not work as expected in Windows due to the differences in line ending.

msoheem committed:
Updated version of Mitre enterprise techniques

msoheem committed:
Update buildattck.py

Adding the updated version of Mitre enterprise attack techniques

msoheem committed:
Update buildattck.py

adding reference

humpalum commented:
Looks good, thank you.

I'll finish reviewing and update the extension end of December early January when I'm back home from vacation :)

Thanks a lot for your contribution!

msoheem commented:
Happy vacation 🙂
Thank you!

msoheem committed:
Update buildattck.py

adding tactics, software, datasource, campaign, mitigations, groups

msoheem committed:
Add files via upload

adding mitre files.

msoheem committed:
Update buildattck.py

humpalum commented:
![image](https://github.com/user-attachments/assets/413a7049-a3e0-4bdd-8a37-300d7d01cfac)
The Python is not working with the given csv's.

Could you please check that out and fix it?

msoheem commented:
This does not happen on my computer. I believe the reason is that on your system the csv's are saved in different encoding because i got this error when i changed the encoding. 
You can change the encoding using VS-Code by following the below steps as I can not default it.
1. Open file with vscode.
2. Click on the encoding on the right down corner.
3. Click on "Save with encoding".
4. Use "utf-8".
![image](https://github.com/user-attachments/assets/8f63a885-a66a-439f-833c-a1c8c5dc8467)

Sorry for the late reply.

humpalum commented:
Sorry for taking so long. I've fixed the encodings and merged it in another PR :)

Thanks for contributing!

msoheem commented:
Thank you 🙂

Yahoo Mail: Search, Organize, Conquer 
 
  On Wed, Mar 26, 2025 at 10:48 AM, Tobias ***@***.***> wrote:    
Sorry for taking so long. I've fixed the encodings and merged it in another PR :)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>
humpalum left a comment (humpalum/vscode-sigma#31)
Sorry for taking so long. I've fixed the encodings and merged it in another PR :)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>
